integration/solana/simple_collectible.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// DISCLAIMER: This file is an example of how to mint and transfer NFTs on Solana. It is not production ready and has not been audited for security.
+// Use it at your own risk.
+
+import '../../solana-library/spl_token.sol';
+
+contract SimpleCollectible {
+    // tokenCounter creates a unique ID for each new minted NFT
+    uint32 private tokenCounter;
+    // The authority responsible for managing mints.
+    address private mintAuthority;
+
+    struct NFTOwner {
+        bool exists;
```

Comment:
What is the purpose of this flag, it looks like there is no code path in this contract to actually set this to false?

Reply:
If I retrieve a key that is not set in the mapping, `exists` is going to be false. This serves to tell me if this item is in the mapping.

Reply:
I guess it's fine if there is no other way to tell (e.g. without using extra space).

Reply:
This is an oddity of how Solidity implements mappings. They are not iterable, and any key will give you an result, but if the entry doesn't exist all the fields will be 0/false.

Reply:
My point was more whether the flag is actually needed, since it uses extra storage space. Where as checking other fields _
might do the same job (e.g. asset URI or addresses etc. can't be 0 anyways).